### PR TITLE
fix(iOS): Image cache module not longer sets null in NSCache

### PR DESCRIPTION
**Pull request by MCurran16**
## PR Checklist

- [x] The PR title follows our guidelines: https://github.com/NativeScript/NativeScript/blob/master/CONTRIBUTING.md#commit-messages.
- [x] There is an issue for the bug/feature this PR is for. To avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it.
- [x] You have signed the [CLA](http://www.nativescript.org/cla).
- [x] All existing tests are passing: https://github.com/NativeScript/NativeScript/blob/master/DevelopmentWorkflow.md#running-unit-tests.
- [x] Tests for the changes are included - https://github.com/NativeScript/NativeScript/blob/master/WritingUnitTests.md.

## What is the current behavior?

If the image-cache module cannot download the provided url (ex. nothing there, expired AWS url, etc.), then it crashes the app with the following error:

```
Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '-[NSCache setObject:forKey:cost:]: attempt to insert nil value
```

## What is the new behavior?

If the NSData returned from `UIImage.alloc().initWithData` is `nil`, do not set it in NSCache.
The error will be passed back to the user for the specified key:

```
this.imageCache_.push({
   key: url,
   url: url,
   completed: (key, image) => {
      console.log('all is well')
      // use image as usual
   },
   error: (key) => {
       console.log('Error or no result for key: ', key);
   }
});
```

Fixes (iOS side of)  #7033 .

### CODE REVIEW

**Comment by MCurran16** [comment]
I'm not exactly sure why the `verification/cla-signed` bot is complaining. I have my email configured in my `git config`. Any advice on how to fix that?

Thanks!

**Comment by NickIliev** [comment]
@MCurran16 make sure that the primary email[ in the Github settings ](https://github.com/settings/emails)is the one set up in your local GitHub config.

**Comment by manoldonev** [comment]
test

**Comment by MCurran16** [comment]
@NickIliev Thank you for your help! Fixed

File: tests/app/ui/image-cache/image-cache-tests.ts [hunk 1]
```diff
@@ -38,3 +36,38 @@ export function test_DummyTestForSnippetOnly() {
     cache.disableDownload();
     // << image-cache-request-images
 }
+
+export const test_ImageCache_NothingAtProvidedUrl = function() {
```
Comment (MartoYankov) [inline tests/app/ui/image-cache/image-cache-tests.ts]: You will have to rework this test. You copied it from the dummy snippet above and it isn't much of a test right now.
1. Remove all the unneeded code for the test. You don't need the `enableDownload`, `disableDownload` and `cache.get()` parts here. Only he `maxRequests` and `cache.push()` parts must stay.
2. Use a boolean var that would indicate whether the cache errored out and assign it a value of `true` in the error callback.
3. Use the `TKUnit.waitUntilReady()` mechanic to wait for the boolean var to become `true`. See how this is done in another test here - https://github.com/NativeScript/NativeScript/blob/master/tests/app/ui/animation/css-animation-tests.ts#L362
4. Assert the error message using `TKUnit.assertEqual` for example.

Let me know if you need help with that.

Comment (MCurran16) [reply tests/app/ui/image-cache/image-cache-tests.ts]: Tests have been improved. Thanks!

File: tests/app/ui/image-cache/image-cache-tests.ts [hunk 2]
```diff
@@ -1,25 +1,23 @@
 ﻿// >> image-cache-require
 import * as imageCacheModule from "tns-core-modules/ui/image-cache";
 import * as imageSource from "tns-core-modules/image-source";
-import * as fs from "tns-core-modules/file-system";
 // << image-cache-require
 
-export function test_DummyTestForSnippetOnly() {
+export const test_DummyTestForSnippetOnly = function() {
```
[the remark this answers is not in the document]
Comment (MCurran16) [reply tests/app/ui/image-cache/image-cache-tests.ts]: Fixed

**Comment by MartoYankov** [comment]
test

**Comment by MartoYankov** [comment]
test

**Comment by SvetoslavTsenov** [comment]
test ios#rc

**Pull request merged**